interpreter/interpreter_test.go

```diff
@@ -528,335 +790,66 @@ func TestInterpreter_SetProto2PrimitiveFields(t *testing.T) {
 	}
 }
 
-func TestInterpreter_GetObjectEnumField(t *testing.T) {
-	src := common.NewTextSource("a.repeated_nested_enum[0]")
-	parsed, errors := parser.Parse(src)
-	if len(errors.GetErrors()) != 0 {
-		t.Errorf(errors.ToDisplayString())
-	}
-
-	pkgr := packages.NewPackage("google.expr.proto3.test")
-	reg := types.NewRegistry(&proto3pb.TestAllTypes{})
-	env := checker.NewStandardEnv(pkgr, reg)
-	env.Add(decls.NewIdent("a", decls.NewObjectType("google.expr.proto3.test.TestAllTypes"), nil))
-	checked, errors := checker.Check(parsed, src, env)
-	if len(errors.GetErrors()) != 0 {
-		t.Errorf(errors.ToDisplayString())
-	}
-
-	i := NewStandardInterpreter(pkgr, reg, reg)
-	eval, _ := i.NewInterpretable(checked)
-	a := &proto3pb.TestAllTypes{
-		RepeatedNestedEnum: []proto3pb.TestAllTypes_NestedEnum{
-			proto3pb.TestAllTypes_BAR,
-		},
-	}
-	vars, _ := NewActivation(map[string]interface{}{
-		"a": reg.NativeToValue(a),
-	})
-	result := eval.Eval(vars)
-	expected := int64(1)
-	got, ok := result.(ref.Val).Value().(int64)
-	if !ok {
-		t.Fatalf("cannot cast result to int64: result=%v", result)
-	}
-	if !reflect.DeepEqual(got, expected) {
-		t.Errorf("Could not build object properly. Got '%v', wanted '%v'",
-			result.(ref.Val).Value(),
-			expected)
-	}
-}
-
-func TestInterpreter_SetObjectEnumField(t *testing.T) {
-	// Test the use of enums within object construction, and their equivalence
-	// int values within CEL.
-	src := common.NewTextSource(
-		`TestAllTypes{
-			repeated_nested_enum: [
-				0,
-				TestAllTypes.NestedEnum.BAZ,
-				TestAllTypes.NestedEnum.BAR],
-			repeated_int32: [
-				TestAllTypes.NestedEnum.FOO,
-				TestAllTypes.NestedEnum.BAZ]}`)
-	parsed, errors := parser.Parse(src)
-	if len(errors.GetErrors()) != 0 {
-		t.Errorf(errors.ToDisplayString())
-	}
-
-	pkgr := packages.NewPackage("google.expr.proto3.test")
-	reg := types.NewRegistry(&proto3pb.TestAllTypes{})
-	env := checker.NewStandardEnv(pkgr, reg)
-	checked, errors := checker.Check(parsed, src, env)
-	if len(errors.GetErrors()) != 0 {
-		t.Errorf(errors.ToDisplayString())
-	}
-
-	i := NewStandardInterpreter(pkgr, reg, reg)
-	eval, _ := i.NewInterpretable(checked, Optimize())
-	expected := &proto3pb.TestAllTypes{
-		RepeatedNestedEnum: []proto3pb.TestAllTypes_NestedEnum{
-			proto3pb.TestAllTypes_FOO,
-			proto3pb.TestAllTypes_BAZ,
-			proto3pb.TestAllTypes_BAR,
-		},
-		RepeatedInt32: []int32{
-			int32(0),
-			int32(2),
-		},
-	}
-	result := eval.Eval(EmptyActivation())
-	got, ok := result.(ref.Val).Value().(*proto3pb.TestAllTypes)
-	if !ok {
-		t.Fatalf("cannot cast result to int64: result=%v", result)
-	}
-	if !reflect.DeepEqual(got, expected) {
-		t.Errorf("Could not build object properly. Got '%v', wanted '%v'",
-			result.(ref.Val).Value(),
-			expected)
-	}
-}
-
-func TestInterpreter_ConstantReturnValue(t *testing.T) {
```

Comment:
Missing.

Reply:
Replaced by `literal_<type>` tests.

Reply:
There's no literal_int.

Reply:
Sure, but the test being replaced is testing a constant return, and there are now several tests that do this.

Reply:
Ok, so the difference is intentional.  That's fine.